File: JavaScript/01.JavaScript-Basic-February-2021/14.exams/2019.05.02_03-Exam/4.touristShop.js
function touristShop(input) {
    index = 0
    let budget = input[index];
    index++;

    let command = input[index];
    index++

    let total = 0;
    let count = 0;
    while (command != "Stop") {
        if (total > budget) {
            break;
        }

        let name = command;
        let price = Number(input[index]);
        index++;
        count++;

        if (count % 3 == 0) {
            price = price * 0.5;
        }
        total += price;

        command = input[index];
        index++
    }

    if (command == "Stop") {
        console.log(`You bought ${count} products for ${total.toFixed(2)} leva.`);
    }

    if (total > budget) {
        console.log(`You don't have enough money!`);
        console.log(`You need ${(total - budget).toFixed(2)} leva!`);
    }
}

touristShop(["153.20", "Backpack", "25.20", "Shoes", "54", "Sunglasses", "30", "Stop"]);
touristShop(["54", "Thermal underwear", "24", "Sunscreen", "45"]);

File: Python/01.Python-Basic-April-2019/15.exams/exam_2019.05.02/4.tourist_shop.py
budget = float(input())

counter = 0
total = 0
command = input()
while command != 'Stop':
    product_name = command
    price = float(input())

    counter += 1
    if counter % 3 == 0:
        price = 0.5 * price

    total += price

    command = input()

rest = abs(budget - total)
if total > budget:
    print(f'You don\'t have enough money!')
    print(f'You need {rest:.2f} leva!')
else:
    print(f'You bought {counter} products for {total:.2f} leva.')
